reference/cd-gitlab-ci.md

```diff
@@ -0,0 +1,183 @@
+by your repository, GitLab has the concept of [Protected Branches and Tags](https://gitlab.com/help/user/project/protected_branches.md).
+
+If you are running `pulumi` from any branch other than the `master` branch,
+you are likely to hit an error that the `PULUMI_ACCESS_TOKEN` 
+environment variable (introduced later in this document) cannot be accessed.
+You can fix this by specifying a wildcard regex to allow specific branches to
+be able to access the secret environment variables.
+
+## Merge Request Builds
+GitLab CI currently does not yet support creating pipelines specifically for merge requests.
+Because of this, pipelines will be created for every single branch that you push up.
+You can apply some amount of control with the use of the configuration variables
+`only` and `except`. You may also consider using `only:changes`, however,
+note that there are some caveats to that. You can learn more [here](https://docs.gitlab.com/ee/ci/yaml/#only-changes).
+
+See [this](https://gitlab.com/gitlab-org/gitlab-ce/issues/23902) issue for community edition and 
+[this](https://gitlab.com/gitlab-org/gitlab-ee/issues/7380) for enterprise edition to learn more.
+
+## Environment Variables
+To use Pulumi within Travis CI, there are a few environment variables you'll need to set for each
+build.
+
+The first is `PULUMI_ACCESS_TOKEN`, which is required to authenticate with pulumi.com in order to
+perform the preview or update. You can create a new Pulumi access token specifically for your
+CI/CD job on your [Pulumi Account page](https://app.pulumi.com/account/tokens).
+
+Next, you will also need to set environment variables specific to your cloud resource provider.
+For example, if your stack is managing resources on AWS, `AWS_ACCESS_KEY_ID` and
+`AWS_SECRET_ACCESS_KEY`.
+
+**Note**: It is a good security practice to mark any sensitive variables as protected in GitLab.
+Please read the note about **Protected Branches** above.
+
+## Scripts
+
+Your repository must contain the `.gitlab-ci.yml` in the root. GitLab looks there by default.
+If you are using an alternate location, be sure to update the settings for your GitLab project
+by going to https://gitlab.com > (select your project) > Settings > General.
+
+The following are samples only. You may choose to structure your configuration any way you like.
+The `setup.sh` installs the `pulumi` CLI on the GitLab CI Runner, and other tools.
+It also installs `yarn` and `nodejs` since that's the runtime for this sample project.
+
+The `run-pulumi.sh` script runs the `pulumi up` command to apply any stack changes and to start
+updating your infrastructure.
+
+The `pulumi-preview.sh` script (not shown here) is similar to the `run-pulumi.sh`, except that
+it runs the `pulumi preview` command instead of the `pulumi up` command, which is sort of a dry-run
+that only shows you changes (if any) in your infrastructure. Isn't that awesome?
```

Comment:
I personally love the "Isn't that awesome?" sentiment. But it _might_ be a little flavor for documentation.

Reply:
Do you mean, remove it? I am OK to remove it if we would like to keep the docs just strictly formal.

Reply:
I would remove it.  

Primarily because it comes off as obnoxious in a script, especially if someone runs into a problem.  I personally just want a set of steps i can simply follow to get from point A to point B effectively.

Reply:
Well, it is in production right now. I'll remove it.